tests/coffee_script/CoffeeLintBearTest.py

```diff
@@ -11,6 +11,228 @@ class Gangster
+
+bad_file1 = """
+1 || 1
+1 && 1
+1 != 1
+1 == 1
+x = !y
+""".splitlines(keepends=True)
+
+
+good_file2 = """
+# Lint your CoffeeScript!
+
+class Gangster
+
+    wasItAGoodDay : () ->
+        yes
+""".splitlines(keepends=True)
+
+
+good_file3 = """
+x = 1234
+y = 1
+""".splitlines(keepends=True)
+
+
+bad_file3 = """
+x = 1234      \ny = 1
+""".splitlines(keepends=True)
+
+
+good_file4 = """
+x(-> 3)
+x( -> 3)
+""".splitlines(keepends=True)
+
+
+bad_file4 = """
+{x((a,b)-> 3)
+""".splitlines(keepends=True)
+
+
+good_file5 = """
+myFunction(a, b, c)
+""".splitlines(keepends=True)
+
+
+bad_file5 = """
+`myFunction a,b,c
+""".splitlines(keepends=True)
+
+
+good_file6 = """
+x = '1234'; console.log(x)
+{spacing:true}
+""".splitlines(keepends=True)
+
+
+bad_file6 = """
+alert('end of line');
+""".splitlines(keepends=True)
+
+
+good_file7 = """
+foo = (empty=(-> undefined)) -> undefined
+{ a, b }
+{ }
+foo = 'bar'
+{ spacing : true }
+""".splitlines(keepends=True)
+
+
+bad_file7 = """
+-> =>
+blah = () ->
+y++
+--x
+{a, b}
+{}
+foo = "bar"
+{spacing:true}
+""".splitlines(keepends=True)
+
+good_file8 = """
+x++;
+x == 1
+doSomething(foo = ',',bar)
+myFunction a, b, {1:2, 3:4}
+`with(document) alert(height);`
+foo = "#{inter}foo#{polation}"
+@ ok
+is(@).ok
+""".splitlines(keepends=True)
+
+
+bad_file8 = """
+throw 'my error'
+throw "#{1234}"
+throw '''
+  long string
+'''
+foo = '#{inter}foo#{polation}'
+myFunction a, b, 1:2, 3:4
+""".splitlines(keepends=True)
+
+
+good_file9 = """
+class SomeThing
+  getConfig: ->
+    @config =
+      foo: 1
+  @getConfig: ->
+    config =
+      foo: 1
+""".splitlines(keepends=True)
+
+
+bad_file9 = """
+class SomeThing
+  getConfig: ->
+    one = 1
+    one = 5
+    @config =
+      keyA: one
+      keyB: one
+      keyA: 2
+  getConfig: ->
+    @config =
+      foo: 1
+  @getConfig: ->
+    config =
+      foo: 1""".splitlines(keepends=True)
+
+
+good_file10 = """
+@[ok]
+@ok()
+doSomething(foo = ',', bar)\nfooBar()
+c = new Foo 1, 2
+throw 'my error'
+throw "#{1234}"
+""".splitlines(keepends=True)
+
+
+bad_file10 = """
+@ notok
+not(@).ok
+@::
+doSomething(foo = ',',bar)\nfooBar()
+""".splitlines(keepends=True)
+
+
+good_file11 = """
+class Y extends X
+  constructor: ->
+    @.hello
+""".splitlines(keepends=True)
+
+
+bad_file11 = """
+class Y extends X
+  constructor: ->
+    this.hello
+""".splitlines(keepends=True)
+
+
+good_file12 = """
+g = new Foo(1, 2)
+h = new Foo(
+  config: 'parameter'
+)
+i = new bar.foo.Foo(1, 2)
+j = new bar.foo.Foo(
+  config: 'parameter'
+)
+myFunction a, b, 1:2, 3:4
+foo = '#{bar}'
+""".splitlines(keepends=True)
+
+
+bad_file12 = """
+c = new Foo 1, 2
+d = new Foo
+  config: 'parameter'
+e = new bar.foo.Foo 1, 2
+f = new bar.foo.Foo
+  config: 'parameter'
+""".splitlines(keepends=True)
+
+
+good_file13 = """
+class Foo
+  constructor: () ->
+    bla()
+  a: "b"
+  c: "d"
+
+class Bar extends Foo
+  constructor: () ->
+    bla()
+""".splitlines(keepends=True)
+
+
+bad_file13 = """
```

Comment:
could you not number them but try to give them meaningful names? Each of those tests has a purpose and naming can help seeing it.